File: backend/src/app/schemas/account.py
from pydantic import BaseModel, EmailStr, Field
from datetime import datetime


class AccountCreate(BaseModel):
    email: EmailStr
    name: str = Field(min_length=1, max_length=255)
    password: str = Field(min_length=6, max_length=128)


class LoginRequest(BaseModel):
    email: EmailStr
    password: str


class AccountOut(BaseModel):
    id: int
    email: EmailStr
    name: str
    created_at: datetime

File: frontend/src/types/account.ts
export type MeResponse = {
  id: number;
  email: string;
  name: string;
  created_at: string;
};

export type AccountOut = MeResponse;

export type TokenPair = {
  access_token: string;
  refresh_token: string;
  token_type: string;
};
